apps/server/src/stores/EventStore.ts

```diff
@@ -32,10 +32,11 @@ export const eventStore = {
     //if there is already and update pending we don't need to schedule another one
     if (!isUpdatePending) {
       isUpdatePending = setImmediate(() => {
+        const patch: Partial<RuntimeStore> = {};
         for (const dataKey of changedKeys) {
-          socket.sendAsJson({ type: `ontime-${dataKey}`, payload: store[dataKey] });
+          Object.assign(patch, { [dataKey]: store[dataKey] });
```

Comment:
suggestion: as for adding the keys to the object. I think a direct assignment can lead to simpler code and it will also be more performant

All the structures here are read-only so we dont really need the immutability that comes with Object.assign

Reply:
yeah agree, but could not get typescript to like this, might be misunderstanding something
```ts
patch[dataKey] = store[dataKey]
```

Reply:
I think this problem will go away if we adopt the batch
the batch function will receive a key + value and we can just use that like

```ts
patch[dynamicKey] = value
````
in which chase the typing is in the function signature